src/vivarium/component.py

```diff
@@ -581,25 +561,164 @@ def get_initialization_parameters(self) -> Dict[str, Any]:
 
     def build_lookup_tables(self, builder: "Builder") -> None:
         """
-        Method to create standard lookup tables for the component. This will create a
-        lookup table for each lookup key in self.standard_lookup_tables property. If
-        additional lookup tables are desired, users have, users have two options: (1)
-        override this method by calling the super method and adding them, or
-        (2) overriding the standard 'lookup_tables' property.
-        """
-        for lookup_table_name in self.standard_lookup_tables:
-            lookup_table_config = builder.configuration[self.name][lookup_table_name]
-            # TODO: make path to configuration the data key when we align artifact
-            # keys with configuration path
-            if lookup_table_config["value"] == "data":
-                table = builder.lookup.build_table(
-                    data=builder.data.load(lookup_table_config["key_name"]),
-                    key_columns=lookup_table_config["categorical_columns"],
-                    parameter_columns=lookup_table_config["continuous_columns"],
-                )
+        Builds lookup tables for this component.
+
+        This method builds lookup tables for this component based on the data
+        sources specified in the configuration. If no data sources are specified,
+        no lookup tables are built.
+
+        The created lookup tables are stored in the lookup_tables dictionary of
+        the component, with the table name as the key.
+
+        Parameters
+        ----------
+        builder : Builder
+            The builder object used to set up the component.
+
+        Returns
+        -------
+        None
+        """
+        self.get_value_columns = builder.data.value_columns()
+        if (
+            self.name in builder.configuration
+            and "data_sources" in builder.configuration[self.name]
+        ):
+            data_source_configs = builder.configuration[self.name].data_sources
+            for table_name in data_source_configs.keys():
+                try:
+                    self.lookup_tables[table_name] = self.build_lookup_table(
+                        builder, data_source_configs[table_name]
+                    )
+                except ConfigurationError as e:
```

Comment:
So this means we will raise an error if the configuration for table name is wrong OR we will hit an error somewhere in `build_lookup_table `if something were done wrong?

Reply:
The only expected ConfigurationError would be thrown by `get_data` if the data source is invalid.